### DISCUSSION TITLE
How to stack PCB's from multiple KiCad projects

### DISCUSSION THREAD

**Discussion by alextrical** [INTI-CMNB/KiBot#724] [Q&A] [answered]
Im looking to stack up PCB's using the pcb2blender.
However as the kicad projects are seperated into different repositories/.kicad_pcb file is there a way to Merge 2 or 3 different .kicad_pcb file's into a single file

Is there a KiBot tool for this? The closest I can see looks to be a BOM consolidation

**Comment by set-soft** [comment]
I'm working on a pre-flight that could help here. I think tomorrow you'll have some thing to try.

**Reply by alextrical** [reply]
That would be fantastic, looking forward to seeing what you are able to achieve. Thank you for working on this amazing project ^_^

**Reply by set-soft** [reply, marked as answer]
Ok, the example is here: https://github.com/INTI-CMNB/KiBot/tree/dev/docs/samples/Consolidate_PCBs
Try to apply this concept to your project, you need to:

1. Tag the stack positions using the mechanism I explained before
2. Take note of the PCB sizes to create a consolidation template
3. Create the template, big enough to hold all the PCBs, and draw text boxes with aprox. the same size as the PCBs to merge
4. Put a name on each text box
5. Make copy of this template, KiBot will back-up it, but is better if you have a copy
6. Create a KiBot config associating each name with the PCB file
7. Run KiBot to get the consolidated PCB

Once you get a good result you need to put the template in the repo and adapt your workflow to make the consolidation as a preflight.

**Reply by alextrical** [reply]
Looks really good, I will get it tried out tomorrow.
I have been running KiBot locally on a Ubuntu install for testing, but will try the dev build on GitHub actions to test out this workflow.

Thank you so much for getting this functionality added to KiBot Dev ^_^
